test/core.jl

```diff
@@ -6461,6 +6461,13 @@ let A=[0, missing], B=[missing, 0], C=Vector{Union{Int, Missing}}(undef, 6)
     @test isequal(C, [0, missing, missing, missing, 0, missing])
 end
 
+# non-power-of-2 element sizes
+primitive type TypeWith24Bits 24 end
+let x = Core.Intrinsics.trunc_int(TypeWith24Bits, 0x112233), a = [x,x]
+    Core.arrayset(true, a, x, 2)
```

Comment:
`a[2] = x`, or are you intentionally trying to avoid this being compiled (could use a comment)?

Can you add a test for `a = [(x, x), (x, x)]` also?

Might be good to use something other than `x` here also, so that the mutation here is observable.

Reply:
OK. Yes, I was trying to avoid the compiled form since I believe it does the right thing.